src/items/services/item.service.ts

```diff
@@ -101,4 +102,24 @@ export class ItemService {
 
     return item;
   }
+
+  async delistAnItem({ itemId, address }: DelistItemRequestDto): Promise<Item> {
+    const item = await this.itemRepository.findById(itemId);
+    if (!item) throw new NotFoundException(`The item with id ${itemId} does not exist`);
+
+    await this.itemRepository.delistAnItem(itemId);
+
+    const account = await this.accountRepository.findByAddress(address);
```

Comment:
Move this up, before delisting. You have to make sure first if the account exist